Mathlib/Order/JordanHolder.lean

```diff
@@ -696,97 +431,97 @@ theorem snoc_snoc_swap {s : CompositionSeries X} {x₁ x₂ y₁ y₂ : X} {hsat
 
 end Equivalent
 
-theorem length_eq_zero_of_bot_eq_bot_of_top_eq_top_of_length_eq_zero {s₁ s₂ : CompositionSeries X}
-    (hb : s₁.bot = s₂.bot) (ht : s₁.top = s₂.top) (hs₁ : s₁.length = 0) : s₂.length = 0 := by
-  have : s₁.bot = s₁.top := congr_arg s₁ (Fin.ext (by simp [hs₁]))
+theorem length_eq_zero_of_head_eq_head_of_last_eq_last_of_length_eq_zero
+    {s₁ s₂ : CompositionSeries X} (hb : s₁.head = s₂.head)
+    (ht : s₁.last = s₂.last) (hs₁ : s₁.length = 0) : s₂.length = 0 := by
   have : Fin.last s₂.length = (0 : Fin s₂.length.succ) :=
-    s₂.injective (hb.symm.trans (this.trans ht)).symm
+    s₂.injective (hb.symm.trans ((congr_arg s₁ (Fin.ext (by simp [hs₁]))).trans ht)).symm
```

Comment:
If you like you can get rid of the next porting note now!

Reply:
thanks